src/cipher.rs

```diff
@@ -510,296 +511,145 @@ impl rustls_root_cert_store {
-    _private: [u8; 0],
-}
+            let client_verifier_builder: &mut Option<ClientCertVerifierBuilder> = try_mut_from_ptr!(builder);
+            let client_verifier_builder = match client_verifier_builder {
+                None => return AlreadyUsed,
+                Some(v) => v,
+            };
 
-impl CastConstPtr for rustls_allow_any_anonymous_or_authenticated_client_verifier {
-    type RustType = AllowAnyAnonymousOrAuthenticatedClient;
-}
+            client_verifier_builder.allow_anonymous = true;
 
-impl ArcCastPtr for rustls_allow_any_anonymous_or_authenticated_client_verifier {}
+            rustls_result::Ok
+        }
+    }
 
-impl rustls_allow_any_anonymous_or_authenticated_client_verifier {
-    /// Create a new allow any anonymous or authenticated client certificate verifier builder
-    /// from the builder.
-    ///
-    /// The builder is consumed and cannot be used again, but must still be freed.
-    ///
-    /// The verifier can be used in several `rustls_server_config` instances. Must be
-    /// freed by the application when no longer needed. See the documentation of
-    /// `rustls_allow_any_anonymous_or_authenticated_client_verifier_free` for details about lifetime.
-    /// This copies the contents of the `rustls_root_cert_store`. It does not take
-    /// ownership of the pointed-to data.
+    // TODO(@cpu): Write comment.
     #[no_mangle]
-    pub extern "C" fn rustls_allow_any_anonymous_or_authenticated_client_verifier_new(
-        builder: *mut rustls_allow_any_anonymous_or_authenticated_client_builder,
-    ) -> *const rustls_allow_any_anonymous_or_authenticated_client_verifier {
+    pub extern "C" fn rustls_web_pki_client_cert_verifier_builder_build(
+        builder: *mut rustls_web_pki_client_cert_verifier_builder,
+    ) -> *const rustls_client_cert_verifier {
         ffi_panic_boundary! {
-            let client_cert_verifier_builder: &mut Option<AllowAnyAnonymousOrAuthenticatedClient> = try_mut_from_ptr!(builder);
+            let client_verifier_builder: &mut Option<ClientCertVerifierBuilder> = try_mut_from_ptr!(builder);
+            let client_verifier_builder = match client_verifier_builder.take() {
+                None => return null() as *const _,
+                Some(v) => v,
+            };
 
-            let client_cert_verifier = match client_cert_verifier_builder.take() {
-                None => {
-                    return null() as *const _;
-                },
-                Some(x) => x,
+            let mut builder = WebPkiClientVerifier::builder(client_verifier_builder.roots)
+                .with_crls(client_verifier_builder.crls);
+            if client_verifier_builder.allow_anonymous {
+                builder = builder.allow_unauthenticated();
+            }
+
+            let verifier = match builder.build() {
+                Ok(v) => v,
+                Err(_) => return null() as *const _, // TODO(@cpu): handle this better.
```

Comment:
@jsha Is the right way to handle the fallibility of the `build()` operation to rework this function to return a `rustls_result` and use an "out pointer arg" for the builder? :thinking: 

As-is when returning `*const rustls_client_cert_verifier` the best I can do is give back `NULL` when there's an error and it eats up whatever the underlying `Err(err)` was.